File: src/NotificationPortal.Web/Controllers/NotificationsController.cs
using System.Threading.Tasks;
using Akka.Actor;
using Microsoft.AspNetCore.Mvc;
using NotificationPortal.Web.ActorModel;
using NotificationPortal.Web.Models;

namespace NotificationPortal.Web.Controllers
{
    public class NotificationsController : Controller
    {
        private readonly RelogifyActorModel _relogifyActorModel;

        public NotificationsController(RelogifyActorModel relogifyActorModel)
        {
            _relogifyActorModel = relogifyActorModel;
        }

        public async Task<IActionResult> Index()
        {
            var notificationsResponse =
                await _relogifyActorModel.NotificationActor
                    .Ask<GetNotificationsResponse>(new GetNotificationsMessage());

            var notificationsViewModel = new NotificationsViewModel
            {
                Notifications = notificationsResponse.Notifications
            };

            return View(notificationsViewModel);
        }
    }
}

File: src/NotificationPortal.Web/Controllers/NotificationsController.fs
namespace NotificationPortal.Web.Controllers

open System.Collections.Generic
open Microsoft.AspNetCore.Mvc
open NotificationPortal.Data
open Microsoft.EntityFrameworkCore

type NotificationsViewModel = { Notifications: List<ChallengeNotification> }

type NotificationsController (dbContext : ApplicationDbContext) =
    inherit Controller()

    member this.Index () =
        async {
            let! notifications = dbContext.Notifications.ToListAsync() |> Async.AwaitTask
            let notificationsViewModel: NotificationsViewModel = { Notifications = notifications }

            return this.View(notificationsViewModel)
        }
